app/assets/javascripts/discourse/app/components/quote-button.js

```diff
@@ -264,11 +315,56 @@ export default Component.extend({
     );
   },
 
+  _saveFastEditDisabled: propertyEqual(
+    "_fastEditInitalSelection",
+    "_fastEditNewSelection"
+  ),
+
   @action
   insertQuote() {
     this.attrs.selectText().then(() => this._hideButton());
   },
 
+  @action
+  _toggleFastEditForm() {
+    if (this._isFastEditable) {
+      this.toggleProperty("_displayFastEditInput");
+
+      schedule("afterRender", () => {
+        document.querySelector("#fast-edit-input")?.focus();
+      });
+    } else {
+      const postId = this.quoteState?.postId;
+      const postModel = this.topic.postStream.findLoadedPost(postId);
+      this.editPost(postModel);
+    }
+  },
+
+  @action
+  _saveFastEdit() {
+    const postId = this.quoteState?.postId;
+    const postModel = this.topic.postStream.findLoadedPost(postId);
+
+    this.set("_isSavingFastEdit", true);
+
+    return ajax(`/posts/${postModel.id}`, { type: "GET", cache: false })
```

Comment:
This probably does not work well when there is an "edit conflict". I think it is fine for now, but it is something worth keeping in mind.

Reply:
Yes I doubt that would be very frequent though, will see if that causes issues in reality, there's lot of things that could go wrong in this PR, but I want to see if this will actually be the case, this is why this is behind the site setting default to false